File: jetpackcamerax/src/main/java/com/jetpackcamerax/MainActivity.java
package com.jetpackcamerax;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import com.camera.CameraActivity;

public class MainActivity extends AppCompatActivity {


    private ImageView iv;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                cameraPhoto();
            }
        });


        findViewById(R.id.btn_galley).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, AlbumShowActivity.class);
                startActivity(intent);
            }
        });


        iv = findViewById(R.id.iv);


    }


    private void cameraPhoto() {

        Intent intent = new Intent(MainActivity.this, CameraActivity.class);

        startActivityForResult(intent, 100);

    }


    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        switch (requestCode) {
            case 100:

                if (resultCode == RESULT_OK) {
                    String path = data.getStringExtra("path");

                    //file:///storage/emulated/0/Android/media/com.example.cameraxapp/Camera/2020-12-31-17-19-21-184.jpg


                    iv.setImageURI(Uri.parse(path));


                }


                break;
        }
    }


}

File: databinding/src/main/java/com/databinding/MainActivity.kt
package com.databinding

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import com.databinding.databinding.ActivityMainBinding
import com.databinding.demo.DemoActivity
import com.databinding.one.OneWayBindingActivity
import com.databinding.two.TwoWayActivity

class MainActivity : AppCompatActivity() {


    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        var user = User(10, "ZZZ")
        binding.userBean = user

        //手动设置
//        binding.tvName.text = ""


        binding.btnOneWay.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, OneWayBindingActivity::class.java)
            startActivity(intent)
        })

        binding.btnTrans.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, TwoWayActivity::class.java)
            startActivity(intent)

        })

        binding.btnDemo.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, DemoActivity::class.java)
            startActivity(intent)
        })


    }


}
